tests/unit/training/lightning/test_default_trainer.py

```diff
@@ -9,3 +21,257 @@ def test_default_trainer():
         experiment_version="x0",
     )
     assert isinstance(result, pl.Trainer)
+
+
+class MockModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear = torch.nn.Linear(10, 5)
+
+    def forward(self, x):
+        return self.linear(x)
+
+
+@pytest.fixture
+def mock_model():
+    return MockModel()
+
+
+@pytest.fixture
+def mock_trace_input():
+    return torch.randn(1, 10)
+
+
+@pytest.fixture
+def mock_lightning_module():
+    """Fixture providing a mock lightning module."""
+    mock_module = MagicMock()
+    mock_module._modules = {}  # pylint: disable=protected-access
+    return mock_module
+
+
+@pytest.fixture
+def trainer_mocks(mocker):
+    """Fixture providing common mocks for trainer tests."""
+    return {
+        "super_save": mocker.patch.object(pl.Trainer, "save_checkpoint"),
+        "jit_export": mocker.patch(
+            "zetta_utils.training.lightning.trainers.default.jit_trace_export"
+        ),
+        "onnx_export": mocker.patch("zetta_utils.training.lightning.trainers.default.onnx_export"),
+        "is_global_zero": mocker.patch.object(
+            ZettaDefaultTrainer, "is_global_zero", new_callable=PropertyMock
+        ),
+        "lightning_module": mocker.patch.object(
+            ZettaDefaultTrainer, "lightning_module", new_callable=PropertyMock
+        ),
+    }
+
+
+def test_save_checkpoint_calls_exports_when_enabled(
+    trainer_mocks, mock_model, mock_trace_input, mock_lightning_module
+):
+    """Test that save_checkpoint calls export functions when exports are enabled."""
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        trainer = ZettaDefaultTrainer(
+            experiment_name="unit_test",
+            experiment_version="x0",
+            enable_jit_export=True,
+            enable_onnx_export=True,
+            default_root_dir=tmp_dir,
+        )
+
+        trainer.trace_configuration = {
+            "test_model": {"model": mock_model, "trace_input": (mock_trace_input,)}
+        }
+
+        filepath = os.path.join(tmp_dir, "test_checkpoint.ckpt")
+
+        trainer_mocks["is_global_zero"].return_value = True
+        trainer_mocks["lightning_module"].return_value = mock_lightning_module
+
+        trainer.save_checkpoint(filepath)
+
+        trainer_mocks["onnx_export"].assert_called_once_with(
+            mock_model, (mock_trace_input,), filepath, "test_model"
+        )
+        trainer_mocks["jit_export"].assert_called_once_with(
+            mock_model, (mock_trace_input,), filepath, "test_model"
+        )
+        trainer_mocks["super_save"].assert_called_once()
+
+
+def test_save_checkpoint_skips_jit_when_disabled(
+    trainer_mocks, mock_model, mock_trace_input, mock_lightning_module
+):
+    """Test that save_checkpoint skips JIT export when disabled."""
```

Comment:
seems not necessary. if jit.trace is disabled in the input param it's clear that it's disabled.